### PR TITLE
Upgrades

### PR DESCRIPTION
**Pull request by JFranklin3000**
Changed Weapon Upgrade to Upgrades 
Added drop down for type of Upgrade
Renamed fields to match the new name

validate Upgrade sheet works correctly.

### CODE REVIEW

File: module/helpers/config.mjs [hunk 1]
```diff
@@ -352,6 +352,13 @@ E20.gearType = {
   tools: "E20.GearTools"
 };
 
+//Upgrade types
+E20.upgradeType = {
+  armor: "E20.UpgradeArmor",
+  drone: "E20.UpgradeDrone",
+  weapon: "E20.UpgradeWeapon"
+};
```
Comment (phildominguez) [inline module/helpers/config.mjs:360]:
```suggestion
// Upgrade types
E20.upgradeType = {
  armor: "E20.UpgradeArmor",
  drone: "E20.UpgradeDrone",
  weapon: "E20.UpgradeWeapon",
};
```

File: lang/en.json [hunk 2]
```diff
@@ -338,6 +338,15 @@
...
     "TransformerEnergon": "Energon",
+    "Upgrade": "Upgrade",
+    "UpgradeArmor": "Armor Upgrade",
+    "UpgradeAvailability": "Availability",
+    "UpgradeBenefit": "Benefit",
+    "UpgradeDrone": "Drone Upgrade",
+    "UpgradePrerequisite": "Prerequisite",
+    "UpgradeType": "Type",
+    "UpgradeWeapon": "Weapon Upgrade",
+    "Upgrades": "Upgrades",
```
Comment (phildominguez) [inline lang/en.json:349]:
```suggestion
    "Upgrade": "Upgrade",
    "UpgradeAvailability": "Availability",
    "UpgradeBenefit": "Benefit",
    "UpgradePlural": "Upgrades",
    "UpgradePrerequisite": "Prerequisite",
    "UpgradeTypeArmor": "Armor",
    "UpgradeTypeDrone": "Drone",
    "UpgradeTypeWeapon": "Weapon",
    "UpgradeType": "Type",
```
Having "Blah Upgrade" for all the types seems redundant to me. I changed `Upgrades` to `UpgradePlural` as well.

File: lang/en.json [hunk 3]
```diff
@@ -196,7 +196,7 @@
     "ItemTypeThreatPowerPlural": "Threat Powers",
     "ItemTypeTraitPlural": "Traits",
     "ItemTypeWeaponPlural": "Weapons",
-    "ItemTypeWeaponUpgradePlural": "Weapon Upgrades",
+    "ItemTypeUpgradeWeaponPlural": "Weapon Upgrades",
```
Comment (phildominguez) [inline lang/en.json:199]: Idt it's being used

**Review by phildominguez — CHANGES_REQUESTED** [review]

File: lang/en.json (hunk 3 above)
Comment (JFranklin3000) [reply lang/en.json]: removed

**Review by phildominguez — APPROVED** [review]

**Pull request merged**